be/src/exec/olap_scan_node.cpp

```diff
@@ -1250,12 +1250,25 @@ void OlapScanNode::transfer_thread(RuntimeState* state) {
 
     state->resource_pool()->release_thread_token(true);
     VLOG(1) << "TransferThread finish.";
-    std::unique_lock<std::mutex> l(_row_batches_lock);
-    _transfer_done = true;
+    {
+        std::unique_lock<std::mutex> l(_row_batches_lock);
+        _transfer_done = true;
+    }
     _row_batch_added_cv.notify_all();
+
+    std::unique_lock<std::mutex> l(_scan_batches_lock);
+    while (_running_thread != 0) {
+        l.unlock();
+        SleepFor(MonoDelta::FromMilliseconds(5));
+        l.lock();
```

Comment:
Is it elegant to use sleep to do multithread sync? why not use a cv to wait here, and nofity after _running_thread--?

Reply:
I don't want use `_scan_batch_added_cv`, it's hard to understand the relationship between `_scan_batch_added_cv` & `_running_thread `. I'll add a new condition variable.